### ISSUE TITLE
[Feature Request] support pull request templates

### ISSUE THREAD

**Issue by celiawaggoner** [integrations/terraform-provider-github#106]
Hi there,

My team adds pull request templates to all of our repos to remind devs to add tests, update docs, use PR title conventions, etc. so I think it'd be helpful to be able to add a pull request template to repos via terraform.

<img width="681" alt="screen shot 2018-07-20 at 9 17 54 am" src="https://user-images.githubusercontent.com/13388602/43013467-d8535128-8bfd-11e8-8ed9-c807413725d9.png">

### Affected Resource(s)
- github_repository

### References
- https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/ 

Thanks for considering!

**Comment by radeksimko** [comment]
Hi @celiawaggoner 
thanks for opening this issue.

Codifying and managing PR/issue templates as code is something we plan to do as well at HashiCorp internally for all of our Terraform providers.

That said I'm not entirely sure whether a resource in the Github provider is the right way to implement it. Managing these templates in practice requires more imperative, rather than declarative approach.

In other words these templates need to be versioned alongside the rest of the codebase and go through the same sequence of (practically irreversible) operations, like `git add`, `git commit`, `git push` and optionally PR.

I will keep this open though, to facilitate discussion, at least until we come up with a way to do this, whether inside or outside of this provider.

**Comment by edahlseng** [comment]
For what it's worth, I've been using https://github.com/edahlseng/terraform-provider-repository-template to keep files like PR templates in sync across several repositories.
